### PR TITLE
Mongo Rbac Swagger Changes

### PR DESCRIPTION
**Pull request by abhijitkaranjkar89**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionDelete.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "parameters": {
+    "userName": "myUserName",
+    "databaseName": "sales",
+    "subscriptionId": "mySubscriptionId",
+    "resourceGroupName": "myResourceGroupName",
+    "accountName": "myAccountName",
+    "api-version": "2021-07-01-preview"
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionDelete.json]:
```suggestion
    "api-version": "2021-10-15-preview"
```

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "parameters": {
+    "userName": "myUserName",
+    "databaseName": "sales",
+    "subscriptionId": "mySubscriptionId",
+    "resourceGroupName": "myResourceGroupName",
+    "accountName": "myAccountName",
+    "api-version": "2021-07-01-preview"
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json]:
```suggestion
    "api-version": "2021-10-15-preview"
```

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/mongorbac.json [hunk 3]
```diff
@@ -0,0 +1,601 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Cosmos DB",
+    "description": "Azure Cosmos DB API for MongoDB Role Based Access Control Resource Provider REST API",
+    "version": "2021-06-15"
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/mongorbac.json]: Should this be 2021-10-15-preview?

File: specification/cosmos-db/resource-manager/readme.md [hunk 4]
```diff
@@ -42,6 +42,7 @@ input-file:
   - Microsoft.DocumentDB/stable/2021-10-15/privateLinkResources.json
   - Microsoft.DocumentDB/stable/2021-10-15/rbac.json
   - Microsoft.DocumentDB/stable/2021-10-15/restorable.json
+  - Microsoft.DocumentDB/stable/2021-10-15/mongorbac.json
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/readme.md]: From the files, I see that you are only adding the changes to the preview version. So this should not be necessary.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionList.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{
+  "parameters": {
+    "subscriptionId": "mySubscriptionId",
+    "resourceGroupName": "myResourceGroupName",
+    "accountName": "myAccountName",
+    "api-version": "2021-07-01-preview"
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionList.json]:
```suggestion
    "api-version": "2021-10-15-preview"
```

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/mySubscriptionId/resourceGroups/myResourceGroupName/providers/Microsoft.DocumentDB/databaseAccounts/myAccountName/mongodbUserDefinitions/myUserId",
+        "name": "myUserName",
```
Comment (kavskalyan) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json]: nit: usually the name is the last of part of the id. I have to check if this is a hard requirement.

**Review by kavskalyan — CHANGES_REQUESTED** [review]

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionDelete.json (hunk 1 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionDelete.json]: Updated.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json (hunk 2 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json]: Updated.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/mongorbac.json (hunk 3 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/mongorbac.json]: Updated.

File: specification/cosmos-db/resource-manager/readme.md (hunk 4 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/readme.md]: Updated.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionList.json (hunk 5 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionList.json]: Updated.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json (hunk 6 above)
Comment (abhijitkaranjkar89) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-10-15-preview/examples/CosmosDBMongoDBUserDefinitionGet.json]: Updated.

**Review by kavskalyan — APPROVED** [review]

**Pull request merged**